### PR TITLE
Use std::semaphore for UnshackledMutex

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

We couldn't use it early because of apple clang

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: lib/Basics/UnshackledMutex.h [hunk 1]
```diff
@@ -33,13 +33,12 @@ namespace arangodb::basics {
 // This mutex lifts this requirement and can be unlocked from another thread.
 class UnshackledMutex {
```
Comment (gnusi) [inline lib/Basics/UnshackledMutex.h:34]: do we still need a wrapper?

Comment (MBkkt) [reply lib/Basics/UnshackledMutex.h:34]: Yes?
See _operationMutex usage in https://github.com/arangodb/arangodb/blob/devel/arangod/Replication2/ReplicatedLog/LogCore.cpp

**Review by maierlars — APPROVED** [review]

File: lib/Basics/UnshackledMutex.h (hunk 1 above)
Comment (joerg84) [reply lib/Basics/UnshackledMutex.h:34]: @gnusi is this question resolved?

Comment (gnusi) [reply lib/Basics/UnshackledMutex.h:34]: @joerg84 yes

**Pull request merged**